lib/Recur/RRuleIterator.php

```diff
@@ -1230,4 +1238,103 @@ protected function getMonths(): array
 
         return $recurrenceMonths;
     }
+
+    private function getNextIterationDateTime(\DateTimeImmutable|\DateTime $dateTime, string $modify): \DateTimeImmutable|\DateTime
+    {
+        $startTs = $dateTime->getTimestamp();
+        $initialDateTime = clone $dateTime;
+        $modifiedDateTime = $dateTime->modify($modify);
+
+        $this->detectAndSaveDstOffset(
+            $initialDateTime,
+            $modifiedDateTime,
+            $modify,
+        );
+
+        $modifiedDateTime = $this->revertPastOffset($modifiedDateTime);
+
+        return $modifiedDateTime;
+    }
+
+    private function detectAndSaveDstOffset(
+        \DateTimeImmutable|\DateTime $initialDateTime,
+        \DateTimeImmutable|\DateTime $modifiedDateTime,
+        string $modify,
+    ) {
+        $dstTransitionLeap = $this->getDstTransitionLeap($initialDateTime, $modifiedDateTime, $modify);
+        if (!is_null($dstTransitionLeap)) {
+            $this->leapOffset[$this->counter] = ($this->leapOffset[$this->counter] ?? 0) + $dstTransitionLeap;
+        }
+
+    }
+
+    private function getDstTransitionLeap(
+        \DateTimeImmutable|\DateTime $initialDateTime,
+        \DateTimeImmutable|\DateTime $modifiedDateTime,
+        string $modify,
+    ): ?int {
+        $modifiedDateInterval = $this->substractDates($modifiedDateTime, $initialDateTime);
+        $modifyInterval = \DateInterval::createFromDateString($modify);
```

Comment:
```suggestion
        $modifyInterval = $modifiedDateTime->sub($initialDateTime);
```

unless I'm missing some detail? :sweat_smile:

Reply:
My first implementation was using `diff` but I noticed it has an odd behaviour on one specific date-time :grimacing: see: https://onlinephp.io/c/57e27

Reply:
ouch, annoying